contrib/ruby/ext/trilogy-ruby/cext.c

```diff
@@ -1202,6 +1210,9 @@ RUBY_FUNC_EXPORTED void Init_cext(void)
     Trilogy_EOFError = rb_const_get(Trilogy, rb_intern("EOFError"));
     rb_global_variable(&Trilogy_EOFError);
 
+    Trilogy_AuthPluginError = rb_const_get(Trilogy, rb_intern("AuthPluginError"));
+    rb_global_variable(&Trilogy_AuthPluginError);
```

Comment:
```suggestion
    rb_global_variable(&Trilogy_AuthPluginError);
    Trilogy_AuthPluginError = rb_const_get(Trilogy, rb_intern("AuthPluginError"));
```

Global variables should be registered before they are assigned. The windows for a bug is extremely small, so most gems get away with it, but we should correct that.